packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -69,29 +76,39 @@ export class BlobManager {
         });
     }
 
+    private hasBlob(id: string): boolean {
+        return this.blobIds.has(id) || this.pendingBlobIds.has(id) ||
```

Comment:
I do not see detached blobs being added to this.blobIds(), and thus making logic inconsistent. For example, blobCount() would not properly count them. I also think you should not check this.pendingBlobIds, as there should be no way for callers to have such blobId until it lands into this.blobIds. I think same is true for this.redirectTable (for somewhat similar reasons)

Reply:
They are added in setRedirectTable(), but you are right about this.pendingBlobIds/this.redirectTable